server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/SetMessagesCreationProcessorTest.java

```diff
@@ -119,106 +164,137 @@ public void processShouldReturnNonEmptyCreatedWhenRequestHasNonEmptyCreate() thr
+        SetMessagesCreationProcessor sut = new SetMessagesCreationProcessor(
+                mockSessionMapperFactory, mimeMessageConverter, mockedMailSpool, mockedMailFactory, messageFactory, fakeSystemMailboxesProvider) {
             @Override
-            protected MessageWithId<Message> createMessageInOutboxAndSend(MessageWithId.CreationMessageEntry createdEntry, MailboxSession session, Mailbox outbox, Function<Long, MessageId> buildMessageIdFromUid) {
-                return new MessageWithId<>(createdEntry.getCreationId(), FAKE_MESSAGE);
-            }
-            @Override
-            protected Optional<Mailbox> getOutbox(MailboxSession session) throws MailboxException {
-                Mailbox fakeOutbox = mock(Mailbox.class);
-                when(fakeOutbox.getName()).thenReturn("outbox");
-                return Optional.of(fakeOutbox);
+            protected MessageWithId createMessageInOutboxAndSend(ValueWithId.CreationMessageEntry createdEntry, MailboxSession session, Mailbox outbox, Function<Long, MessageId> buildMessageIdFromUid) {
+                return new MessageWithId(createdEntry.getCreationId(), FAKE_OUTBOX_MESSAGE);
             }
         };
         // When
-        SetMessagesResponse result = sut.process(buildFakeCreationRequest(), buildStubbedSession());
+        SetMessagesResponse result = sut.process(createMessageInOutbox, session);
 
         // Then
         assertThat(result.getCreated()).isNotEmpty();
         assertThat(result.getNotCreated()).isEmpty();
     }
 
-    @Test(expected = MailboxRoleNotFoundException.class)
-    public void processShouldThrowWhenOutboxNotFound() {
+    @Test
+    public void processShouldReturnErrorWhenOutboxNotFound() {
         // Given
-        SetMessagesCreationProcessor sut = new SetMessagesCreationProcessor(null, null, null, null, null, null, messageFactory) {
-            @Override
-            protected Optional<Mailbox> getOutbox(MailboxSession session) throws MailboxException {
-                return Optional.empty();
-            }
-        };
+        TestSystemMailboxesProvider doNotProvideOutbox = new TestSystemMailboxesProvider(Optional::empty, () -> optionalDrafts);
+        SetMessagesCreationProcessor sut = new SetMessagesCreationProcessor(
+                stubSessionMapperFactory, mimeMessageConverter, mockedMailSpool, mockedMailFactory, messageFactory, doNotProvideOutbox);
         // When
-        sut.process(buildFakeCreationRequest(), null);
+        SetMessagesResponse actual = sut.process(createMessageInOutbox, session);
+        
+        assertThat(actual.getNotCreated()).hasSize(1).containsKey(creationMessageId);
+        assertThat(actual.getNotCreated().get(creationMessageId).getType()).isEqualTo("error");
     }
 
     @Test
     public void processShouldCallMessageMapperWhenRequestHasNonEmptyCreate() throws MailboxException {
```

Comment:
nothing is really needed here as most is already tested by integration testing, but It's useful nonetheless because it's a strong requirement (store something) and feedback loop is shorter